### ISSUE TITLE
Changing subscriber options on bulk

### ISSUE THREAD

**Issue by bobinoz** [websharks/comment-mail#327]
My host's email service, Mailchannels, banned me for sending out too many spam messages. This, of course, is a very serious issue and I never send out spam, but I can see why people are marking it as such. Some of them signed up for these notifications five years ago, no wonder they have forgotten.

My settings, by the way, default to do not subscribe as I don't think it is fair or right to automatically subscribe somebody to anything. Unfortunately, when people do subscribe themselves, they almost always choose the replies to all comments option.

Under the StCR plug-in, I could tidy this up quite easily and I used to do so on a regular basis. I would click on manage subscriptions from the front end of a page or post and change everyone who had subscribed more than one year ago to only receive replies to their comments rather than replies to all comments. It was very quick and easy to do.

Under the new Comment Mail, this not only appears to be impossible to do on bulk, but I cannot even do it individually even if I had enough time to do so. I can delete them completely, but that really does defeat the object of having the plug-in in the first place.

So, is there a way to change subscriber options on bulk in either lite or pro versions? Or even better, is there a way of finding out who is complaining of spam and therefore deleting them from my system? I noticed the pro version has some kind of automatic cleanup of subscribers built in, would that help me in my situation in any way?

Many thanks in advance for any help you might be able to give, Bob

**Comment by raamdev** [comment]
@bobinoz The way to do this right now is with the Pro version, where you can use the Import/Export tools to do a mass export, which generates a CSV file, then make whatever changes are desired, and re-import that CSV to update the subscribers. See [CSV Import/Export Tools](http://comment-mail.com/kb-article/csv-importexport-tools/).

----

That said, I agree that it would be nice if the Subscriptions list had a few more options bulk changes (I'm thinking how the Posts lists allows you to do things like bulk-change the author of a set of posts, or the post status, etc.). Right now, the only bulk actions are related to subscription status, and options for deleting/trashing:

![2016-12-14_12-19-36](https://cloud.githubusercontent.com/assets/53005/21192796/bfa05e5e-c1f7-11e6-9438-0ad0592f8860.png)

I'll mark this as an enhancement.

----

> Or even better, is there a way of finding out who is complaining of spam and therefore deleting them from my system? I noticed the pro version has some kind of automatic cleanup of subscribers built in, would that help me in my situation in any way?

The Subscription Cleaner automatically deletes unconfirmed and trashed subscriptions. It's not possible for Comment Mail to detect when somebody marks a notification as spam, because that notification doesn't come to Comment Mail at all. Where it goes depends on the email client the user is using (e.g., if it's Gmail, the spam complaint goes to Google).

**Comment by jaswrks** [comment]
I agree that having bulk actions, other than changing the status, would be nice. To start with, the desire that @bobinoz has for changing the subscription type would be nice.

**Comment by bobinoz** [comment]
The CSV option seems a little clunky, so when you say @raamdev you are marking it as an enhancement, does that mean you're hopeful of introducing a simpler bulk subscription type change option in a future release?

**Comment by jaswrks** [comment]
@bobinoz Yes, I think that is the consensus here; i.e., we agree that adding additional Bulk Actions to that list would be very helpful in a future release.
